### ISSUE TITLE
pilot-agent response  in multiple threads

### ISSUE THREAD

**Issue by yankunsam** [istio/istio#44423]
### Bug Description

I use ebpf to trace kprobe, ```tcp_sendmsg```. And it shows that about 6 or 13 threads response for "/stats/prometheus" the same time. 
![i3bKuRAh9d](https://user-images.githubusercontent.com/16130147/232713445-857d4621-bfbf-4ddb-a178-b60d2158e525.jpg)

<img width="954" alt="image" src="https://user-images.githubusercontent.com/16130147/232717467-bc4322e5-51b4-4a6e-9688-562f7eee2152.png">

### Version

```prose
istio v1.10.1
```

### Affected product area

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [X] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by hzxuzhonghu** [comment]
What is 100MB mean?

**Comment by yankunsam** [comment]
@hzxuzhonghu It means size in ebpf map field is large than 100MB, 
```c
else {
			trafficp->sent += size;
			
			
			if (trafficp->sent > 100000000) {
			    bpf_printk("update send > 100MB occured: %d \n",trafficp->sent);
				bpf_printk("saddr %u \n",ip_key.saddr);
				bpf_printk("daddr %u \n",ip_key.daddr);
			    bpf_printk("family: %d \n",family);
				bpf_printk("sport: %d\n",ip_key.lport);
				bpf_printk("dport: %d\n",ip_key.dport);
				bpf_printk("update size: %d \n",size);
			}
		}
```

**Comment by kebe7jun** [comment]
stats/promethues are sidecar metrics, in general, he will increase with the business access topology, for the general system, the amount of data output each time may be more than 300kb, according to the general case, 15s per collection frequency, in 300 times (4500s) may reach 100m, and your The ebpf program does not distinguish between pods, so if there are multiple pods, this amount of data may be reached faster, but theoretically this should have little effect.

**Comment by hzxuzhonghu** [comment]
I doubt the scraping from envoy is too slow because of the large amount of metrics. So if the scraping period is shorter than the time cost. There could be multi concurrent threads handling stats

**Comment by yankunsam** [comment]
@kebe7jun 1. we use vmagent to pull metrics from pod every 30s. 2. I will delete the key of  ebpf map every 1s, so no 300 times. 3. pod ip as one field of ebpf map key, so multiple pods no effect.

**Comment by yankunsam** [comment]
@hzxuzhonghu I have a test in my local minikube cluster, there are also multiple concurrent threads handling stats. BTW, when trace the receive function like ```kprobe/tcp_cleanup_rbuf```, the count is equal the metics bytes.

**Comment by kebe7jun** [comment]
You can see how big these pods are for you on a single occasion(curl pod up:15090/stats/prometheus). I'm not quite clear on the logic of how your ebpf clears the data, not sure if that includes cumulative data.

**Comment by howardjohn** [comment]
There have been many improvements since 1.10 here btw, it is far EOL. Worth upgrading before debugging more.

**Comment by howardjohn** [comment]
Unfortunately the Istio project cannot reasonably support infinite versions though. If you cannot upgrade that is unfortunate but its also not reasonable to expect help from the community - we could easily be wasting our time with bugs we already fixed years ago if we spent time on EOL versions. 

You don't need to upgrade your production, but unless the issue is reproducible in a supported version I think this will be closed.
